### PR TITLE
Improve AOT Compilation Accuracy.

### PR DESCRIPTION
**Pull request by apghml**

### CODE REVIEW

File: axlearn/common/launch.py
Comment (markblee) [inline axlearn/common/launch.py:17]: How come `infer_tpu_type` (called from `default_xla_options`) doesn't raise when `instance_type="gpu"`? LMK what I missed.

Comment (apghml) [reply axlearn/common/launch.py]: Ah, I totally misunderstood your comment. Fixed by only calling it if the instance type starts with tpu.

Comment (apghml) [reply axlearn/common/launch.py]: The latest version has also been testing on GPU now.

**Comment by apghml** [comment]
@markblee I think I need your approval again before I can merge.

**Review by markblee — APPROVED** [review]

**Pull request merged**
